src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -830,6 +832,33 @@ GraphPatternOperation Visitor::visitPathQuery(
   return pathQuery;
 }
 
+GraphPatternOperation Visitor::visitSpatialQuery(
+    Parser::ServiceGraphPatternContext* ctx) {
+  auto parseSpatialQuery = [ctx](parsedQuery::SpatialQuery& spatialQuery,
+                                 const parsedQuery::GraphPatternOperation& op) {
+    if (std::holds_alternative<parsedQuery::BasicGraphPattern>(op)) {
+      spatialQuery.addBasicPattern(
+          std::get<parsedQuery::BasicGraphPattern>(op));
+    } else if (std::holds_alternative<parsedQuery::GroupGraphPattern>(op)) {
+      spatialQuery.addGraph(op);
+    } else {
+      reportError(
+          ctx,
+          "Unsupported element in spatialQuery."
+          "spatialQuery may only consist of triples for configuration"
+          "And a { group graph pattern } specifying the right join table.");
+    }
+  };
+
+  parsedQuery::GraphPattern graphPattern = visit(ctx->groupGraphPattern());
+  parsedQuery::SpatialQuery spatialQuery;
+  for (const auto& op : graphPattern._graphPatterns) {
+    parseSpatialQuery(spatialQuery, op);
+  }
+
+  return spatialQuery;
```

Comment:
If you
1. Also do the checking here (does my magic service have all required parameters etc.)
2. Wrap everything in a `try {...} cach(all kinds of exceptions) { reportError(ctx, ...}`

Then we get better error messages (the reportError function marks the faulty part of the query red in the UI), this can't be done if the exceptions are thrown only in the query planner or even later.


The same goes of course for the PathSearch.

Reply:
For the spatial search I have implemented this. The PathSearch's check requires a vocabulary which I doubt we have available at this stage in the parsing.